app/scripts/models/flow.js

```diff
@@ -0,0 +1,130 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * A model to represent the somewhat nebulous concept of an
+ * "event flow". A single flow could be a user connecting a
+ * new device to their Sync account, for instance.
+ *
+ * Each flow has two properties.
+ *
+ *   * flowId: 32-byte identifier hex-encoded as a string.
+ *   * flowBeginTime: server-generated timestamp.
+ *
+ * When a new flow is begun, it should result in an activity
+ * event being logged by the server. To that end, we push a
+ * a `flow.begin` onto the metrics event array.
+ *
+ * There is some mess surrounding when to begin a new flow,
+ * which necessitates equivalent mess in this model. It runs
+ * something like this:
+ *
+ *   1. A user arriving at /signin or /signup constitutes the
+ *      beginning of a flow, so a corresponding `flow.begin`
+ *      event must be logged.
+ *
+ *   2. A user refreshing the page on /signin or /signup does
+ *      not constitute a new flow, so events must not be logged
+ *      in those cases. This is achieved by saving the flowId
+ *      in local storage.
```

Comment:
> > A user refreshing the page on /signin or /signup does not constitute a new flow
>
> It's not obvious to me whether this should or should not indicate a new flow. Do you have concrete scenarios in mind for which this is or is not the right thing?

Ha, so earlier on in this PR's history, this was not the case. But it came up in the feedback that we don't want somebody who opens the content server in, say, 3 different tabs and completes the sign up process in 1, to generate 3 separate `flow.begin` events. Or if someone sits there refreshing the page for whatever reason, we wouldn't want that emitting a bunch of events. Otherwise, it would throw out the calculation of the completion percentage, which relies on the overall count of `flow.begin` in the feature spec.

/cc @vladikoff, who IIRC brought this up earlier in the review.